add support for imx8mp-lpddr4-evk

mike-scott opened the pull request:
Patches for:
- u-boot-fio
- optee-os-fio
- boot scripts
- mfgtool files
- machine custom changes
- fstab addition

vanmaegima reviewed (approved):
LGTM

MrCry0 reviewed (approved):
LGTM

Tim-Anderson reviewed (approved):
LGTM

ricardosalveti commented on meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio/imx8mp-lpddr4-evk/lmp-base.cfg:
```diff
@@ -0,0 +1,25 @@
+CONFIG_CMD_EDITENV=y
+CONFIG_CMD_SAVEENV=y
+CONFIG_USE_BOOTCOMMAND=y
+CONFIG_BOOTCOMMAND="fatload mmc ${emmc_dev}:1 ${loadaddr} /boot.scr; source ${loadaddr}; reset"
```
We might not need to define bootcommand here as usually the board has enough to find and use a boot.scr file, but fine to keep if you want.

mike-scott replied:
Yeah most likely this was copied over from imx8mm, but we're a bit inconsistent:
```
./meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio/imx8mm-lpddr4-evk/lmp-base.cfg:26:CONFIG_BOOTCOMMAND="fatload mmc ${emmc_dev}:1 ${loadaddr} /boot.scr; source ${loadaddr}; reset"
./meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio/imx7ulpea-ucom/lmp-base.cfg:52:CONFIG_BOOTCOMMAND="mmc rescan; fatload mmc ${mmcdev}:1 ${loadaddr} boot.scr; source ${loadaddr}"
./meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio/imx8qm-mek/lmp-base.cfg:33:CONFIG_BOOTCOMMAND="fatload mmc 0:1 ${scriptaddr} /boot.scr; source ${scriptaddr}"
./meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio/imx8mq-evk/lmp-base.cfg:25:CONFIG_BOOTCOMMAND="fatload mmc 0:1 ${loadaddr} /boot.scr; source ${loadaddr}; reset"
```

mike-scott replied:
For now, i'll keep it in and we can clean up all of the machines at once if we want to drop this.

ricardosalveti commented on meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio/imx8mp-lpddr4-evk/lmp.cfg:
```diff
@@ -0,0 +1,110 @@
+CONFIG_FIT_SIGNATURE_STRICT=y
+CONFIG_FIT_VERBOSE=y
+CONFIG_SECONDARY_BOOT_RUNTIME_DETECTION=y
+# TODO: fix hard-coded seconday boot offset
```
seconday/secondary

mike-scott replied:
Fixing

ricardosalveti commented on meta-lmp-bsp/recipes-security/optee/optee-os-fio-mfgtool_3.10.0.bbappend:
```diff
@@ -1,26 +1,33 @@
-    CFG_IN_TREE_EARLY_TAS=fiovb/22250a54-0bf1-48fe-8002-7b20f1c9c9b1 \
+EXTRA_OEMAKE_append_apalis-imx8 = " \
+    CFG_NXP_CAAM=n CFG_RNG_PTA=y CFG_UART_BASE=0x5a070000 \
+    CFG_DT=y CFG_EXTERNAL_DTB_OVERLAY=y CFG_DT_ADDR=0x83200000 \
```
Wont't some of these settings be already included by the SoC settings line you added above?

mike-scott replied:
The only generic SoC settings are for `mx8m` which won't apply to apalis-imx8 (imx8qm)

ricardosalveti commented on meta-lmp-bsp/recipes-security/optee/optee-os-fio_3.10.0.bbappend:
```diff
@@ -1,27 +1,40 @@
-    CFG_IN_TREE_EARLY_TAS=fiovb/22250a54-0bf1-48fe-8002-7b20f1c9c9b1 \
+EXTRA_OEMAKE_append_apalis-imx8 = " \
+    CFG_NXP_CAAM=n CFG_RNG_PTA=y CFG_UART_BASE=0x5a070000 \
+    CFG_DT=y CFG_EXTERNAL_DTB_OVERLAY=y CFG_DT_ADDR=0x83200000 \
```
Same here.

mike-scott replied:
The only generic SoC settings are for `mx8m` which won't apply to apalis-imx8 (imx8qm)

ricardosalveti commented on meta-lmp-bsp/recipes-security/optee/optee-os-fio_3.10.0.bbappend:
```diff
@@ -34,29 +47,19 @@ EXTRA_OEMAKE_append_imx7ulpea-ucom = " \
-    CFG_DT=y CFG_EXTERNAL_DTB_OVERLAY=y CFG_DT_ADDR=0x43200000 \
+EXTRA_OEMAKE_append_imx8qm-mek = " \
+    CFG_NXP_CAAM=n CFG_RNG_PTA=y \
+    CFG_DT=y CFG_EXTERNAL_DTB_OVERLAY=y CFG_DT_ADDR=0x83200000 \
```
And here.

mike-scott replied:
The only generic SoC settings are for `mx8m` which won't apply to apalis-imx8 (imx8qm)

ricardosalveti commented on meta-lmp-bsp/recipes-security/optee/optee-os-fio_3.10.0.bbappend:
```diff
@@ -1,27 +1,40 @@
 OPTEEMACHINE_imx8mm-lpddr4-evk = "imx-mx8mmevk"
+OPTEEMACHINE_imx8mq-evk = "imx-mx8mqevk"
+OPTEEMACHINE_imx8qm-mek = "imx-mx8qmmek"
+OPTEEMACHINE_qemuarm64 = "vexpress-qemu_armv8a"
```
Nit, but I would prefer the non functional changes to be separated in another commit, easier to review and revert if needed later on in case of regressions.

mike-scott replied:
Yep.  Splitting this into 2 commits.

ricardosalveti commented on meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc:
```diff
@@ -360,6 +360,35 @@ BOOTSCR_LOAD_ADDR_imx8mm-lpddr4-evk = "0x44800000"
+## iMX8MP EVK
+OSTREE_KERNEL_ARGS_imx8mp-lpddr4-evk ?= "console=tty1 console=ttymxc1,115200 earlycon=ec_imx6q,0x30890000,115200 root=/dev/mmcblk2p2 rootfstype=ext4"
+SOTA_CLIENT_FEATURES_append_imx8mp-lpddr4-evk = " ubootenv"
+MACHINE_FEATURES_remove_imx8mp-lpddr4-evk = "qca9377 qca6174"
```
Not sure if this line is needed, wasn't able to find these features in the machine conf.

mike-scott replied:
Yep, this looks like a copy pasta issue.  Removing.

ricardosalveti commented on meta-lmp-bsp/dynamic-layers/freescale-layer/recipes-bsp/imx-mkimage/imx-boot_%.bbappend:
```diff
@@ -4,8 +4,16 @@ DEPENDS_remove = "optee-os"
+SRC_URL_append_mx8mq = " \
      file://0001-iMX8M-support-SPL-ddr-sign.patch \
+"
+
```
No need for this extra line.

ricardosalveti commented on meta-lmp-bsp/dynamic-layers/freescale-layer/recipes-bsp/imx-mkimage/imx-boot_%.bbappend:
```diff
@@ -4,8 +4,16 @@ DEPENDS_remove = "optee-os"
 
 
-SRC_URI_append_mx8m = " \
+# TEMP: Need to figure out why this is breaking mfgtool builds on mx8mp
```
Maybe replace TEMP with TODO? Usually easier to identify stuff we need to work on later on.

Also fine to remove [TEMP] from the commit subject.

mike-scott replied:
Fixing.

ricardosalveti commented:
@mike-scott @Tim-Anderson looks good, some minor comments and nits.

One thing that would be nice to avoid is having patches without a commit message, several only got the subject. Even if used to repeat the core of the subject line, useful to add if stuff was based on another board/target or similar, to add a bit more context.

mike-scott commented:
- Rebased on latest `master`
- Split out the non-functional changes in the 2 OP-TEE clean up patches.
- Addressed all comments except imx8qm machine questions in OP-TEE recipe clean ups (this is not inherited from the mx8m SoC).

Diff from previous version:
```
diff --git a/meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc b/meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc
index f4c57d2d..53a305d3 100644
--- a/meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc
+++ b/meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc
@@ -385,7 +385,6 @@ IMXBOOT_TARGETS_mx8mp = "flash_evk_spl"
 ## iMX8MP EVK
 OSTREE_KERNEL_ARGS_imx8mp-lpddr4-evk ?= "console=tty1 console=ttymxc1,115200 earlycon=ec_imx6q,0x30890000,115200 root=/dev/mmcblk2p2 rootfstype=ext4"
 SOTA_CLIENT_FEATURES_append_imx8mp-lpddr4-evk = " ubootenv"
-MACHINE_FEATURES_remove_imx8mp-lpddr4-evk = "qca9377 qca6174"
 UBOOT_MACHINE_imx8mp-lpddr4-evk = "imx8mp_evk_config"
 BOOTSCR_LOAD_ADDR_imx8mp-lpddr4-evk = "0x44800000"

diff --git a/meta-lmp-bsp/dynamic-layers/freescale-layer/recipes-bsp/imx-mkimage/imx-boot_%.bbappend b/meta-lmp-bsp/dynamic-layers/freescale-layer/recipes-bsp/imx-mkimage/imx-boot_%.bbappend
index fa87ac7f..00e21c9d 100644
--- a/meta-lmp-bsp/dynamic-layers/freescale-layer/recipes-bsp/imx-mkimage/imx-boot_%.bbappend
+++ b/meta-lmp-bsp/dynamic-layers/freescale-layer/recipes-bsp/imx-mkimage/imx-boot_%.bbappend
@@ -4,7 +4,7 @@ DEPENDS_remove = "optee-os"
 DEPENDS += "${@bb.utils.contains('MACHINE_FEATURES', 'optee', 'virtual/optee-os', '', d)}"


-# TEMP: Need to figure out why this is breaking mfgtool builds on mx8mp
+# TODO: Need to figure out why this is breaking mfgtool builds on mx8mp
 # For now only apply the patch to mx8mm and mx8mq
 SRC_URL_append_mx8mm = " \
      file://0001-iMX8M-support-SPL-ddr-sign.patch \
@@ -12,7 +12,6 @@ SRC_URL_append_mx8mm = " \
 SRC_URL_append_mx8mq = " \
      file://0001-iMX8M-support-SPL-ddr-sign.patch \
 "
-
 SRC_URI_append_mx8m = " \
      file://0002-iMX8M-add-SPL-only-build.patch \
      file://0003-iMX8M-add-support-for-packing-HDMI-fw-in-SPL-only-bo.patch \
diff --git a/meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio/imx8mp-lpddr4-evk/lmp.cfg b/meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio/imx8mp-lpddr4-evk/lmp.cfg
index bd912863..aa7f330c 100644
--- a/meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio/imx8mp-lpddr4-evk/lmp.cfg
+++ b/meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio/imx8mp-lpddr4-evk/lmp.cfg
@@ -34,7 +34,7 @@ CONFIG_FIT_SIGNATURE=y
 CONFIG_FIT_SIGNATURE_STRICT=y
 CONFIG_FIT_VERBOSE=y
 CONFIG_SECONDARY_BOOT_RUNTIME_DETECTION=y
-# TODO: fix hard-coded seconday boot offset
+# TODO: fix hard-coded secondary boot offset
 # defined in fuse 0x490 [22:19] IMG_CNTN_SET1_OFFSET
 # default value is 0x0 == 4MB offset
 CONFIG_SECONDARY_BOOT_SECTOR_OFFSET=0x1000
```

mike-scott commented:
Adding a note from @rsalveti here: we will want a new PR to combine the settings for the `mx8qm` SoC.  (resolving those questions in this PR)
